### PR TITLE
[echarts] fix series-bar.itemStyle.color for linear gradient & radial gradient

### PR DESCRIPTION
**Pull request by TokenRanOut**
**documents just give a example for series-line,but series-bar works well.**

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://echarts.apache.org/en/option.html#series-line.itemStyle.color>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by bilalucar** [comment]
The url you specify has the color feature of the line chart. The title says bar graph. It is also related to the line graph in the change. Is there a mistake?

**Comment by PlayByMyself** [comment]
@bilalucar documents just give a example for series-line,but series-bar works well.

File: types/echarts/options/series/bar.d.ts [hunk 1]
```diff
@@ -922,7 +922,15 @@ declare namespace echarts {
...
-                color?: string;
+                color?: string | {
+                    type: 'linear',
+                    x: zrender.X,
+                    y: zrender.Y,
+                    x2: zrender.X2,
+                    y2: zrender.Y2,
+                    colorStops: zrender.ColorStops,
+                    globalCoord: zrender.GlobalCoords
+                };
```
Comment (Ovilia) [inline types/echarts/options/series/bar.d.ts]: If the linear gradient is supported, radial should also be supported.

**Review by Ovilia — CHANGES_REQUESTED** [review]

**Review by Ovilia — APPROVED** [review]
Seems great!

**Comment by Ovilia** [comment]
Although this PR is perfect by itself, gradient color is actually avaialble for all color options. This can serve as an improvement for later PRs.

**Comment by PlayByMyself** [comment]
@Ovilia In fact, I want to make it looks like this:
`color?: string | zrender.LinearGradient | zrender.RadialGradient;`
but LinearGradient is a constructor,and no RadialGradient in [zrender](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2988e824bc1615621e805ee699b307d1ed76080b/types/zrender/index.d.ts).
what about improve zrender first,and then make it available for all color options?

**Comment by Ovilia** [comment]
Sure.

**Pull request merged**